### ISSUE TITLE
[Mosquitto] Existing users have no access after installing addon, need to recreate them

### ISSUE THREAD

**Issue by Calimerorulez** [home-assistant/addons#2597]
### Describe the issue you are experiencing

Hi,

When having existing users in HA, and then installing the Mosquitto addon, that users do not have access to the broker.

I'm installing Frigate, so I added the frigate user to HA. Then I installed the Mosquitto addon, but the frigate user does not have access (not authorized). When deleting the user, and recreating it, the user does have access to the broker.

Maybe it's by design.

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Other (e.g., Raspbian/Raspberry Pi OS/Fedora)

### Which add-on are you reporting an issue with?

Mosquitto broker

### What is the version of the add-on?

6.1.2

### Steps to reproduce the issue

1. Create HA user 'frigate'
2. Install and enable Mosquitto addon, and configure the integration.
3. User 'frigate' does not have access.
4. Remove and recreate the 'frigate' user.
5. User has access to the broker.
...

**Comment by mdegat01** [comment]
I just tried exactly your steps and it worked without issue. I created a `frigate` user, installed the addon, started it and now I am connected to it as `frigate`.

Can you share your config? Also can you share how you actually tested the connection? You shouldn't have to enter any user credentials to configure the integration in HA, it should be discovered and set up for you. So where are you putting those credentials?

**Comment by Calimerorulez** [comment]
The user I'm using is configured in Frigate itself, it isn't the user from the Frigate HA addon. I've got a separate installation of Frigate on another server. As you know, Frigate connects via the configured user/pass in its config file.

I created the frigate user in HA as an account with local access only, at first (for MQTT access purposes). Then I installed the Mosquitto addon. Then fired up my Frigate installation. The log of Mosquitto states an access denied for the frigate user, coming from the IP adres of my Frigate server installation. When I remove the HA frigate user, and then recreate it with the same user/pass, the frigate user is allowed to connect.

It's not a big problem by the way :)

I think it is because of the fact I'm not using the Frigate addon in HA, but a separate one on another server. It looks like Mosquitto only picks up users created after its installation.

**Comment by mdegat01** [comment]
> As you know, Frigate connects via the configured user/pass in its config file.

I don't know, I have never used Frigate.

> I think it is because of the fact I'm not using the Frigate addon in HA

Impossible. This makes no difference. Once an addon is running its just a docker container. And in this picture Frigate (regardless of where it is running) is simply an MQTT client. Plus there isn't an official Frigate addon so how would the Mosquitto addon even know about it to do something special with it? Looking at [addon analytics](https://analytics.home-assistant.io/addons.json) there's like 20 different frigate addons people are using with HA, and those are just the ones installed in systems with reporting turned on.

As I said above, I tested exactly what you described. I created a user in HA, then I installed the addon, then I tried to connect to my broker with those credentials. I was not using any addon to do this I was using random 3rd party software for debugging MQTT (MQTT Explorer). There were no issues.

I would suggest you check the credentials you put in Frigate and ensure you put the same username and password in Frigate that you entered in HA. If not then I'm not really sure but I would ask you try and reproduce it with something other then Frigate since it may be an issue with Frigate itself (although I'm not really sure how that is possible either).
